pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -231,6 +231,10 @@ func (alloc *allocateAction) Execute(ssn *framework.Session) {
 
 		if ssn.JobReady(job) {
 			stmt.Commit()
+		} else if ssn.JobCandidateReady(job) {
```

Comment:
I think that's not necessary to add a new callback; that should ok to check best-effort pods in `JobReadyFn`.

Reply:
If the judgement of the best-effort pod is merged into the `JobReady` fuction, when `JobReady` is satisfied, the dispatch of non-best-effort pod will be started, which will cause the non-best-effort pod of the job to be bound to the node first. But at this time, the gang constraint of the job is not satisfied. Maybe the remaining best-effort pods under the job can not be successfully scheduled during the `backfill` action.

Reply:
IMO, we should dispatch all pods of job in allocate action, including best-effort pod :)